modules/core/src/main/java/org/apache/ignite/internal/processors/query/QueryEngine.java

```diff
@@ -40,6 +40,17 @@
     List<FieldsQueryCursor<List<?>>> query(@Nullable QueryContext ctx, String schemaName, String qry, Object... params)
         throws IgniteSQLException;
 
+    /**
+     * @param ctx Query context, may be null.
+     * @param schemaName Schema name.
+     * @param qry Query.
+     * @param batchedParams Optional query parameters.
+     * @return Query cursor.
```

Comment:
Let's describe here, that we return list of query cursors (one cursor per batch).